tests/ui/typeck/irrefutable-while-let-pattern.stderr

```diff
@@ -0,0 +1,31 @@
+error[E0308]: mismatched types
+  --> $DIR/irrefutable-while-let-pattern.rs:4:5
+   |
+LL |   fn foo() -> bool {
+   |               ---- expected `bool` because of return type
+LL | /     while let x = 0 {
+LL | |
+LL | |         return true;
+LL | |     }
+   | |_____^ expected `bool`, found `()`
+   |
+   = note: `while` loops evaluate to unit type `()`
+note: It is determined that this might iterate zero times, regardless of the loop condition expression or pattern.
+  --> $DIR/irrefutable-while-let-pattern.rs:4:5
+   |
+LL |     while let x = 0 {
+   |     ^^^^^^^^^^^^^^^
+help: If you are assuming that it will iterate at least once, consider using a `loop` expression instead.
+  --> $DIR/irrefutable-while-let-pattern.rs:4:5
+   |
+LL |     while let x = 0 {
+   |     ^^^^^^^^^^^^^^^
+help: consider returning a value here
+   |
+LL ~     }
+LL +     /* `bool` value */
+   |
+
+error: aborting due to 1 previous error
```

Comment:
A number of issues here.
- Error messages don't start with an upper-case letter, and don't end with a '.'.
- There is a lot of repetition here. The `note` and the following `help` both print the loop condition. The two could be merged. I am also not sure if the condition needs to be printed, given that the loop is printed above anyway.
- "It is determined" is too wordy, something like "this loop might iterate zero times" would be better.

But there are bigger concerns, which I will discuss below.